### ISSUE TITLE
Warning ``alignContent === 'left' is not supported on this direction.``

### ISSUE THREAD

**Issue by marwie** [felixmariotto/three-mesh-ui#168]
Hello,

have a look at [this fiddle](https://jsfiddle.net/739p8tba/)

I'm trying to align content in the top left corner using the following settings:

```
    contentDirection: "row",
    justifyContent: "start",
    alignContent: "left",
```

the warning ``alignContent === 'left' is not supported on this direction.`` says ([as documented](https://github.com/felixmariotto/three-mesh-ui/wiki/API-documentation)) this is not supported yet it works?
And when changing ``left`` to ``top`` I'm getting a bunch of errors and nothing is rendered anymore:

```
<a class='gotoLine' href='#"THREE.BufferGeometry.computeBoundingSphere(): Computed radius is NaN. The \&quot;position\&quot; attribute is likely to have NaN values.", [object Object'>"THREE.BufferGeometry.computeBoundingSphere(): Computed radius is NaN. The \&quot;position\&quot; attribute is likely to have NaN values.", [object Object</a> {
  addAttribute: function(name, attribute) {
  console.warn("THREE.BufferGeometry: .addAttribute() has been renamed to .setAttribute().");
  if (!(attribute && attribute.isBufferAttribute) && !(attribute && attribute.isInterleavedBufferAttribute)) {
    console.warn("THREE.BufferGeometry: .addAttribute() now expects ( name, attribute ).");
    return this.setAttribute(name, new BufferAttribute(arguments<a class='gotoLine' href='#1], arguments[2'>1], arguments[2</a>));
  }
  if (name === "index") {
    console.warn("THREE.BufferGeometry.addAttribute: Use .setIndex() for index attribute.");
    this.setIndex(attribute);
    return this;
  }
  return this.setAttribute(name, attribute);
},
[...]
```
Left
![image](https://user-images.githubusercontent.com/5083203/157971284-dcfe17c9-7a70-4717-861f-7be54bb0874b.png)
Right
![image](https://user-images.githubusercontent.com/5083203/157972970-b1f6f19a-60ae-4b5e-aecb-7a807d6ad9c3.png)

--- 
```
    contentDirection: "row",
    justifyContent: "start",
    alignContent: "top",
```
![image](https://user-images.githubusercontent.com/5083203/157972283-c90043f3-8c5d-42de-b3af-e1b784aff89a.png)

**Comment by swingingtom** [comment]
Hey @marwie,
It might be a bit confusing atm, but it works as "designed/expected". 

If you look at this [forked fiddle](https://jsfiddle.net/swingingtom/3pzL02t5/12/), same basis as your, replaced the TextComponent for 2 BlockComponents. In that case, if you play with `alignContent` values; we can  see `"top"`,`"bottom"` and `"center"` are valid for `contentDirection : "row"` and are working correctly.

What is the issue with you fiddle? The `alignContent` property currently serves two purposes in three-mesh-ui. 

1. Flex items alignment ( css flex align-items )
2. Inline alignment ( css text-align kind of )

So as Inline alignment (text-align) is currently also done by `alignContent` property. If you set `"top"` on the Block, the warning comes from the Text child unable to text-align top. If you set `"left"` on the parent, the parent cannot work because of invalid value of the flex layout align items. The idea is to follow [nested_blocks ](https://github.com/felixmariotto/three-mesh-ui/blob/15e24494bb1f0fe045e50687ed1e2ec7d9123f6d/examples/nested_blocks.js#L154) example, and add an intermediary block to act as text-align only. See a [fixed fiddle ](https://jsfiddle.net/swingingtom/segdzprx/21/)

Hopefully you'll understand that explanation...

> ... but it works as "designed/expected". 

It doesn't mean it has to/ will stay like it. 

- #158 has deprecated `alignContent` values : `left`,`top`,`bottom`,`right` in favor of more abstracted `start` and `end` values. Which stay valid when switching `contentDirection`.
- We could split `alignContent` to : 
    - `alignItems` ( flex naming ) and `textAlign` ( major API break ) 
    - keeping `alignContent` ( a bit less API break ) and add `textAlign` ?
    - you name it...

**Comment by marwie** [comment]
Ah I see thanks for clarifying! 
Makes sense - looking forward to the PR being merged and will add the intermediate block until then.

**Comment by marwie** [comment]
Would it be possible to provide some context when printing warnings like that (e.g. UUID of the object that has invalid settings)?

**Comment by marwie** [comment]
Thanks for the help, intermediate container helped getting rid of the warnings and thanks to your no-wrap PR we can now marvel at horizontal-overflow in my ugly test scene:
![image](https://user-images.githubusercontent.com/5083203/158027340-08c02537-adc6-4938-9b81-ec696300a307.png)

**Comment by swingingtom** [comment]
Should be fixed with **v6.4.0+** as `alignContent` is now deprecated in favor of `alignItems`.     

`alignItems` values : left,top,bottom,right are also deprecated in favor of more abstracted `start` and `end` values. __Which stay valid when switching `contentDirection`.__

It also have now `textAlign` with align text component see [TextAlign example](https://felixmariotto.github.io/three-mesh-ui/#text_align)
